src/Migrate/custom/New-AzMigrateServerReplication.ps1

```diff
@@ -464,6 +514,251 @@ public static int hashForArtifact(String artifact)
         if ($HasTargetAVZone) {
             $ProviderSpecificDetails.TargetAvailabilityZone = $TargetAvailabilityZone
         }
+
+        if ($HasSqlServerLicenseType)
+        {
+            $validLicenseSpellings = @{ 
+                NoLicenseType = "NoLicenseType";
+                PAYG = "PAYG";
+                AHUB = "AHUB"
+            }
+            $SqlServerLicenseType = $validLicenseSpellings[$SqlServerLicenseType]
+            $ProviderSpecificDetails.SqlServerLicenseType = $SqlServerLicenseType
+        }
+
+        if ($HasTag) {
+            $IllegalCharKey = New-Object Collections.Generic.List[String]
+            $ExceededLengthKey = New-Object Collections.Generic.List[String]
+            $ExceededLengthValue = New-Object Collections.Generic.List[String]
+
+            if ($Tag.Count -gt 50)
+            {
+                throw "InvalidTags : Too many tags specified. Requested tag count - '$($VMTag.Count)'. Maximum number of tags allowed - '50'."
+            }
+
+            foreach ($x in $Tag.Keys)
+            {
+                 if ($x.length -gt 512)
+                 {
+                     $ExceededLengthKey.add($x)
+                 }
+
+                 if ($x -match "[<>%&\?/.]")
+                 {
+                     $IllegalCharKey.add($x)
+                 }
+
+                 if ($($Tag.Item($x)).length -gt 256)
+                 {
+                     $ExceededLengthValue.add($($Tag.Item($x)))
+                 }
+            }
+
+            if ($IllegalCharKey.Count -gt 0)
+            {
+                throw "InvalidTagNameCharacters : The tag names '$($IllegalCharKey -join ', ')' have reserved characters '<,>,%,&,\,?,/' or control characters. These characters are only allowed for tags that start with the prefix 'hidden, link'."
```

Comment:
characters are only allowed for tags that start with the prefix 'hidden, link'."
What is hidden, link?

Reply:
Removed.